src/main.ts

```diff
@@ -450,6 +451,42 @@ export default class ImageWindow extends Plugin {
             })
         );
 
+        this.registerDomEvent(document, "contextmenu", (event: MouseEvent) => {
```

Comment:
Did you explore using this from the obsidian API instead of a DOM event?

https://github.com/obsidianmd/obsidian-api/blob/583ba39e3f6c0546de5e5e8742256a60e2d78ebc/obsidian.d.ts#L4272

Reply:
As far as I can tell, that only works on the context menu in the editing view. My goal was to easily show an image on another screen from a note that's in reading view. (In terms of TTRPGs, I want to quickly show my players an image from a note I have up as a reference.)

Looking at the code again though, I'm not sure if there's a better way to get the cleaned Obsidian path that you might know of. This is my first foray into the Obsidian API.

`            const cleanPath = imgPath.split(`${vaultName}/`)[1].split("?")[0];
`
It works for my case, but I could see there being issues if someone's vault name appears multiple times in the path.

Reply:
No worries! I just looked at it on my phone; I’ll review it in a bit on my computer